File: src/network/packet/impl/ProxyRegisterServerPacket.php
<?php

namespace pocketcloud\cloud\network\packet\impl;

use pocketcloud\cloud\network\client\ServerClient;
use pocketcloud\cloud\network\packet\ClientboundPacket;
use pocketcloud\cloud\network\packet\CloudPacket;
use pocketcloud\cloud\network\packet\util\PacketData;

final class ProxyRegisterServerPacket extends CloudPacket implements ClientboundPacket {

    public function __construct(
        private readonly string $serverName = "",
        private readonly int $port = 0
    ) {}

    public function handle(ServerClient $client): void {}

    public function encodePayload(PacketData $packetData): void {
        $packetData->writeAll($this->serverName, $this->port);
    }

    public function decodePayload(PacketData $packetData): void {}

    public function getServerName(): string {
        return $this->serverName;
    }

    public function getPort(): int {
        return $this->port;
    }

    public static function create(string $serverName, int $port): self {
        return new self($serverName, $port);
    }
}

File: src/main/java/de/pocketcloud/cloud/network/packet/impl/ProxyRegisterServerPacket.java
package de.pocketcloud.cloud.network.packet.impl;

import de.pocketcloud.cloud.network.client.ServerClient;
import de.pocketcloud.cloud.network.packet.ClientboundPacket;
import de.pocketcloud.cloud.network.packet.CloudPacket;
import de.pocketcloud.cloud.network.packet.data.PacketData;
import de.pocketcloud.cloud.server.CloudServer;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.jetbrains.annotations.NotNull;

@NoArgsConstructor
@Getter
public final class ProxyRegisterServerPacket extends CloudPacket implements ClientboundPacket {

    private String serverName;
    private int port;

    public ProxyRegisterServerPacket(String serverName, int port) {
        this.serverName = serverName != null ? serverName : "";
        this.port = port;
    }

    @Override
    public void handle(@NotNull ServerClient client) {}

    @Override
    public void encodePayload(PacketData packetData) {
        packetData.writeAll(serverName, port);
    }

    @Override
    public void decodePayload(PacketData packetData) {}

    public static ProxyRegisterServerPacket create(String serverName, int port) {
        return new ProxyRegisterServerPacket(serverName, port);
    }

    public static ProxyRegisterServerPacket create(CloudServer server) {
        return create(server.name(), server.serverData().port());
    }
}
